classes/Deck.js

```diff
@@ -1,22 +1,187 @@
-import { PlayingCard } from './PlayingCard'
+import { SUITES, RANKS } from '../constants/playing-card'
 
 class Deck {
-  constructor(suites, ranks) {
-    const cards = []
-    for (const suite of suites) {
-      for (const rank of ranks) {
-        cards.push(new PlayingCard(suite, rank))
+  constructor(cards = []) {
+    this.cards = cards
+  }
+
+  validateCanDrawCard(cardPosition) {
+    if ((0 >= cardPosition) || (cardPosition > this.size())) {
```

Comment:
I think it would be better to make sure the card position is not less than 1, since that's what the validation really is.

Also because I don't feel like making sure that `insertCard(someCard, 0.3)`, for example, works as expected since at some point `splice` will be receiving -0.3.